UPGRADE-3.0.md

```diff
@@ -29,4 +35,76 @@ $recommendation = UserRecommendation::create('user-id', 'scenario');
 $recommendation->setCount(5);
 $recommendation->setRotationRate(1.0);
 $recommendation->setRotationTime(3600);
+```
+
+## `Interaction` now accepts interaction type as parameter
+Matej now allows configuration of custom interaction types and interaction attributes.
+
+At the same time, it allows specifying interaction item using item alias instead
+of item ID. For that reason we removed static constructor methods for creating specific interaction types - `Interaction::detailView`, `Interaction::purchase`, `Interaction::bookmark` and `Interaction::rating`.
```

Comment:
What about keep these as aliases for new `Interaction:withItem('detailviews'...` etc.

Reply:
This is not a good idea. Historically we used names on interactions in plural - "ratings", "detailViews" etc. And aliases like `Interaction::rating` point to the plural version. In the future we want to start using singular version of interaction name - "rating", "detailView". New clients (and some older one that we'll change) wouldn't be able to use the alias like `Interaction::rating` to create an interaction name "rating" because it would in fact create an interaction with name in plural. And that's very confusing.

Clients have to change their code anyway, keeping the alias wouldn't save them any time at all. And we are actively developing Matej. We want the api of the client be kept clean for futuru improvements.